components/raftstore-v2/src/operation/command/admin/mod.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+mod conf_change;
+
+use engine_traits::{KvEngine, RaftEngine};
+use kvproto::{
+    raft_cmdpb::{AdminRequest, RaftCmdRequest},
+    raft_serverpb::PeerState,
+};
+use protobuf::Message;
+use raft::prelude::ConfChangeV2;
+use raftstore::{
+    store::{
+        self, cmd_resp,
+        fsm::apply,
+        msg::ErrorCallback,
+        util::{ChangePeerI, ConfChangeKind},
+    },
+    Result,
+};
+use slog::info;
+use tikv_util::box_err;
+
+use self::conf_change::ConfChangeResult;
+use crate::{
+    batch::StoreContext,
+    raft::{Apply, Peer},
+    router::CmdResChannel,
+};
+
+#[derive(Debug)]
+pub enum AdminCmdResult {
+    ConfChange(ConfChangeResult),
+}
+
+impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
+    #[inline]
+    pub fn on_admin_command<T>(
+        &mut self,
+        ctx: &mut StoreContext<EK, ER, T>,
+        mut req: RaftCmdRequest,
+        ch: CmdResChannel,
+    ) {
+        if !self.serving() {
+            apply::notify_req_region_removed(self.region_id(), ch);
+            return;
+        }
+        if let Err(e) = self.validate_command(&req, &mut ctx.raft_metrics) {
+            let resp = cmd_resp::new_error(e);
+            ch.report_error(resp);
+            return;
+        }
+
+        if !self.applied_to_current_term() {
```

Comment:
minor: it would be better to add the comment why applied_to_current_term is needed here.